src/main.rs

```diff
@@ -255,59 +260,166 @@ fn main() {
         .register_inspectable::<MoveInDirection>()
         .register_inspectable::<MoveInArc>()
         .register_inspectable::<Rotate>()
-        .register_inspectable::<Attack>()
+        .register_inspectable::<attack::Attack>()
         .register_inspectable::<YSort>()
         .register_inspectable::<Facing>()
         .register_inspectable::<Panning>();
 
+    // Register assets and loaders
     assets::register(&mut app);
 
     debug!(?engine_config, "Starting game");
 
-    // Insert the game handle
+    // Get the game handle
     let asset_server = app.world.get_resource::<AssetServer>().unwrap();
     let game_asset = engine_config.game_asset;
-    let handle: Handle<Game> = asset_server.load(&game_asset);
-    app.world.insert_resource(handle);
+    let game_handle: Handle<GameMeta> = asset_server.load(&game_asset);
 
+    // Configure hot reload
+    if engine_config.hot_reload {
+        app.add_stage_after(
+            AssetStage::LoadAssets,
+            GameStage::HotReload,
+            SystemStage::parallel(),
+        )
+        .add_system_to_stage(GameStage::HotReload, is_hot_reload.chain(load_game))
+        .add_system_set_to_stage(
+            GameStage::HotReload,
+            ConditionSet::new()
+                .run_in_state(GameState::InGame)
+                .with_system(hot_reload_level)
+                .with_system(hot_reload_fighters)
+                .into(),
+        );
+    }
+
+    // Insert game handle resource
+    app.world.insert_resource(game_handle);
+
+    // Print the graphviz schedule graph
     #[cfg(feature = "schedule_graph")]
     bevy_mod_debugdump::print_schedule(&mut app);
 
     app.run();
 }
 
-fn spawn_camera(mut commands: Commands) {
-    let mut camera_bundle = OrthographicCameraBundle::new_2d();
-    // camera_bundle.orthographic_projection.depth_calculation = DepthCalculation::Distance;
-    camera_bundle.orthographic_projection.scaling_mode = ScalingMode::FixedVertical;
-    camera_bundle.orthographic_projection.scale = 16. * 14.;
-    commands
-        .spawn_bundle(camera_bundle)
-        .insert(Panning {
-            offset: Vec2::new(0., -consts::GROUND_Y),
-        })
-        .insert(ParallaxCameraComponent);
-}
-
+/// Loads the main [`GameMeta`] resource and then transitions to the main menu
+///
+/// This is run in a chain as either `not_hot_reload().chain(load_game)` or
+/// `is_hot_reload().chain(load_game)` so that logic for hot reloading the game and loading the game
+/// can be shared.
 fn load_game(
```

Comment:
Just to make sure I'm understanding correctly, this runs in loading state and then if hot reload is enabled/configured also in a hot reload state with some conditional logic that determines what to do in the hot reload case?

Reply:
Yep, exactly.

Since all the logic was almost exactly the same for both an initial load and hot reload it seemed reasonable to run the same system for both cases and just check an extra is_hot_reload flag.

Granted, it did make it a little more complicated, and I wasn't 100% sure it was worth it. I think it's better than writing most of the same code twice and adding more opportunities for bugs where the hot reload runs different than the initial load, though.

Reply:
It was kindof unintuitive to follow at first but I think it makes sense. I wonder if it could be split up into a `needs_hot_reload()` iyes_loopless run condition which is then applied to a conditionset that runs the hotreload logic in its own system before the load system?

Reply:
could be left for a future pr though if it even makes sense

Reply:
Yeah, I think we could try it layer if we wanted to. I was just analyzing to try and see if it was possible/beneficial.

The issue I see with it is the code that needs to run only when it isn't a hot reload. It'd need to be split up into somewhat more independent systems, which might be a better design anyway, but yeah, I'll probably try that in a separate PR after this one.